inference/inference-ir-trees/src/commonMain/kotlin/io/kinference/trees/SingleModeTreeEnsemble.kt

```diff
@@ -0,0 +1,39 @@
+package io.kinference.trees
+
+import io.kinference.ndarray.arrays.*
+
+abstract class SingleModeTreeEnsemble<T : NumberNDArray>(
+    private val aggregator: Aggregator,
+    protected val transform: PostTransform,
+    private val treeDepths: IntArray,
+    private val treeSizes: IntArray,
+    featureIds: IntArray,
+    nodeFloatSplits: FloatArray,
+    private val nonLeafValuesCount: IntArray,
+    private val leafValues: FloatArray,
+    private val biases: FloatArray,
+    val numTargets: Int,
+    splitMode: TreeSplitType
+) {
+    private val treeSplitter = TreeSplitter.get(splitMode, featureIds, nodeFloatSplits)
+
+    protected fun applyEntry(array: FloatArray, output: FloatArray, srcIdx: Int = 0, dstIdx: Int = 0) {
+        var index: Int
+        var score = FloatArray(numTargets)
+        var treeOffset = 0
+        var off = 0
+        for ((i, depth) in treeDepths.withIndex()) {
+            index = 0
```

Comment:
Do we need `var index: Int` outside of loop? Looks like it must be `var index = 0` inside first loop